types/src/transaction/mod.rs

```diff
@@ -36,11 +36,13 @@ use std::{
 pub mod authenticator;
 mod change_set;
 pub mod helpers;
+mod metadata;
 mod module;
 mod script;
 mod transaction_argument;
 
 pub use change_set::ChangeSet;
+pub use metadata::*;
```

Comment:
Since we are adding a layer of specifications that is relatively independent, I would make the `metadata` module visible as such with `pub mod metadata` instead `pub use metadata::*`. It should also be cleaner in `cargo doc` (see previous comment)